
#-----------------------------------------------------------------------------------------------------------------------
def add(a: int, b: int = 0):
    return a + b


print(add(1))  # не выдает ошибку, так как казано дефолт значение


def add_kw(a: int, b: int, *args, **kwargs):

    return [a, b, args, kwargs]


print(add_kw(1, 2, 4, 6, q=7, c=3, d=4))  # [1, 2, (4, 6), {'q': 7, 'c': 3, 'd': 4}]


print((lambda num: num * 2)(2))  # 4 // ею лучше не злоупотреблять